common/src/main/java/com/wynntils/features/inventory/HightlightDuplicateCosmeticsFeature.java

```diff
@@ -0,0 +1,151 @@
+        // No need to get hovered cosmetic if only displaying selected cosmetics
+        if (highlightCondition.get() == HighlightCondition.SELECTED) return;
+
+        Slot hoveredSlot = event.getHoveredSlot();
+
+        // If no item or on new reward screen don't get a hovered cosmetic, hide §f too as the glass panes on new
+        // reward screen are named that and the return text item isn't added immediately
+        if (hoveredSlot == null
+                || hoveredSlot.getItem().getItem() == Items.AIR
+                || event.getScreen()
+                        .getMenu()
+                        .slots
+                        .get(RETURN_SLOT)
+                        .getItem()
+                        .getHoverName()
+                        .equals(RETURN_TEXT)
+                || hoveredSlot.getItem().getHoverName().equals(Component.literal("§f"))) {
+            hoveredCosmetic = null;
+            return;
+        }
+
+        hoveredCosmetic = hoveredSlot.getItem().getHoverName();
+    }
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Pre e) {
+        if (!onScrapMenu) return;
+
+        // Don't highlight the cosmetics in the selected slots
+        if (SearchableContainerType.SCRAP_MENU.getBounds().getSlots().contains(e.getSlot().index)) {
+            if (selectedCosmetics.contains(e.getSlot().getItem().getHoverName())
+                    && highlightCondition.get() != HighlightCondition.HOVER) {
+                RenderUtils.drawArc(
+                        e.getPoseStack(), selectedHighlightColor.get(), e.getSlot().x, e.getSlot().y, 200, 1f, 6, 8);
+            } else if (e.getSlot().getItem().getHoverName().equals(hoveredCosmetic)
+                    && highlightCondition.get() != HighlightCondition.SELECTED) {
+                RenderUtils.drawArc(
+                        e.getPoseStack(), hoveredHighlightColor.get(), e.getSlot().x, e.getSlot().y, 200, 1f, 6, 8);
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public void onContainerSetContent(ContainerSetContentEvent.Post event) {
+        if (!onScrapMenu) return;
+
+        // Cosmetics have been scrapped and new cosmetic is being displayed, clear selected
+        if (event.getItems().get(RETURN_SLOT).getHoverName().equals(RETURN_TEXT)) {
+            selectedCosmetics.clear();
+            return;
+        }
+
+        // Update the selected cosmetics
+        for (int slot : SELECTED_COSMETIC_SLOTS) {
+            Component selectedCosmetic = event.getItems().get(slot).getHoverName();
+
+            if (selectedCosmetic.equals(ADD_REWARD_TEXT)) {
+                // If the first slot does not have a cosmetic then none are selected
+                if (slot == 1) {
+                    selectedCosmetics.clear();
+                }
+
+                break;
+            }
+
+            if (!selectedCosmetics.contains(selectedCosmetic)) {
+                selectedCosmetics.add(selectedCosmetic);
+            }
```

Comment:
Perhaps selectedCosmetic can be a set? Then you can remove contains. Also, IMO it makes more sense since it represents a set of selected items.